beacon_node/network/src/sync/network_context.rs

```diff
@@ -79,11 +79,18 @@ pub enum RpcResponseError {
 
 #[derive(Debug, PartialEq, Eq)]
 pub enum RpcRequestSendError {
-    /// Network channel send failed
-    NetworkSendError,
-    NoCustodyPeers,
-    CustodyRequestError(custody::Error),
-    SlotClockError,
+    /// No peer available matching the required criteria
+    NoPeer(NoPeerError),
+    /// These errors should never happen, including unreachable custody errors or network send
+    /// errors.
+    InternalError(String),
```

Comment:
I grouped errors with 2 variants as both backfill and forward range need to handle the `NoPeer` case and everything else is treated as an internal error